src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -451,13 +461,18 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
         logger.print("\n[JaCoCo plugin] Saving matched class directories for class-pattern: " + classPattern + ": ");
         for (FilePath file : matchedClassDirs) {
             dir.saveClassesFrom(file);
-            logger.print(" " + file);
```

Comment:
@centic9 Turns out the log message problem was due to a '/' after my class directory name which caused it to iterate over every sub-directory as you mentioned. Proper class directory has reduced the total execution time as well. This was a great help and I appreciate your suggestions. I have added the log messages back as part of this PR.